distributor/cmd/main.go

```diff
@@ -684,9 +684,9 @@ func matchesFilter(e cloudevents.Event) bool {
 	if err := e.DataAs(keptnBase); err != nil {
 		return true
 	}
-	if env.ProjectFilter != "" && keptnBase.Project != env.ProjectFilter ||
-		env.StageFilter != "" && keptnBase.Stage != env.StageFilter ||
-		env.ServiceFilter != "" && keptnBase.Service != env.ServiceFilter {
+	if env.ProjectFilter != "" && !strings.Contains(env.ProjectFilter, keptnBase.Project) ||
```

Comment:
I believe we have to convert the project/stage/service filter to an array and try to find an exact match for `keptnBase.Project`. Otherwise, if we have e.g. a filter with the value:

```
my-project,my-other-project
```

events for projects whose name is a substring of any of those project names would also be forwarded. E.g. if the distributor receives an event with `keptnBase.Project=other` would also fulfill the condition.

Reply:
Absolutely true! Let me change that.